Storageaccount tlsversion

paultaiton opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
This change adds support to azure_rm_storageaccounts.py for the "MinimumTlsVersion" parameter for new and updated storage accounts.
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
azure_rm_storageaccount
##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

paultaiton committed:
update azure storage mgmt package to minimum required for tls_version

paultaiton committed:
minimum sdk version and api version to support minimum_tls_version

paultaiton committed:
no .value properties, sku.tier and sku.name are strings.

paultaiton committed:
added minimum_tls_version parameter for storageaccount

paultaiton commented:
This is the first code of any significance I've written with Azure python SDK and for Ansible modules. I made a lot of assumptions to get this working, so I really appreciate any help with reviewing and getting this to an acceptable quality.

Fred-sun commented:
@paultaiton Thank you for your contribution, we will review and promote PR merger as soon as possible. Thank you!

paultaiton commented:
@Fred-sun 
I have not done ansible tests before, but I'll see what I can throw together and will let you know when I think I have it.
Thanks Fred.

paultaiton committed:
add support for minimum_tls_version

paultaiton commented:
@Fred-sun 
I added some tests, but in the process realized that I did not add sufficient support for the _info module. That has also been added in the latest commit.

During my own testing of adding the TLS feature, I realized that there were errors with the code as it existed, and had to make modifications outside the scope of MinimumTlsVersion to make execution complete without errors.

As mentioned I added some code for the integration tests, but I have no idea how to run it. I THINK it is correct syntactically and logically, but lacking the actual test I cannot confirm. Could I please ask that you either run the test for me, or point me in the right direction for documentation on how to accomplish that myself?

Thanks again for your help.

Fred-sun commented:
> @Fred-sun
> I added some tests, but in the process realized that I did not add sufficient support for the _info module. That has also been added in the latest commit.
> 
> During my own testing of adding the TLS feature, I realized that there were errors with the code as it existed, and had to make modifications outside the scope of MinimumTlsVersion to make execution complete without errors.
> 
> As mentioned I added some code for the integration tests, but I have no idea how to run it. I THINK it is correct syntactically and logically, but lacking the actual test I cannot confirm. Could I please ask that you either run the test for me, or point me in the right direction for documentation on how to accomplish that myself?
> 
> Thanks again for your help.

I will check your change! Thank you very much!

Fred-sun commented on plugins/modules/azure_rm_storageaccount.py:
```diff
@@ -680,6 +694,18 @@ def update_account(self):
                 except Exception as exc:
                     self.fail("Failed to update account type: {0}".format(str(exc)))
 
+        if bool(self.minimum_tls_version) != self.account_dict.get('minimum_tls_version'):
```
```suggestion
        if self.minimum_tls_version is not None and self.minimum_tls_version != self.account_dict.get('minimum_tls_version'):
```

Fred-sun commented on plugins/modules/azure_rm_storageaccount_info.py:
```diff
@@ -517,18 +517,19 @@ def account_obj_to_dict(self, account_obj, blob_service_props=None):
-            account_type=account_obj.sku.name.value,
-            kind=account_obj.kind.value if account_obj.kind else None,
-            provisioning_state=account_obj.provisioning_state.value,
+            account_type=account_obj.sku.name, #.value,
```
```suggestion
            account_type=account_obj.sku.name,
```

Fred-sun reviewed (commented):
@paultaiton Please help change as suggested below. Thank you!

paultaiton committed:
Update plugins/modules/azure_rm_storageaccount.py

Co-authored-by: Fred-sun <37327967+Fred-sun@users.noreply.github.com>

paultaiton committed:
Update plugins/modules/azure_rm_storageaccount_info.py

Co-authored-by: Fred-sun <37327967+Fred-sun@users.noreply.github.com>

Fred-sun commented on plugins/modules/azure_rm_storageaccount.py:
```diff
@@ -94,6 +94,17 @@
+            - TLS1_1
+            - TLS1_2
+        version_added: "2.10"
+        version_added_collection: "azure.azcollection v0.4.0"
```
```suggestion

```

Fred-sun reviewed (commented):
@paultaiton Thanks!

paultaiton committed:
Update plugins/modules/azure_rm_storageaccount.py

Co-authored-by: Fred-sun <37327967+Fred-sun@users.noreply.github.com>

paultaiton commented:
Suggestions look good.

Fred-sun commented:
ready_for_review

paultaiton commented:
Hello @Fred-sun .

Was there any further action required on my part?  I'm not sure what the next steps are.

Fred-sun commented:
> Hello @Fred-sun .
> 
> Was there any further action required on my part? I'm not sure what the next steps are.

Because the maintainer was busy with other things during this time, there are no time to review these PRs. I will push maintainer to handle these PRs today or tomorrow. If it causes you inconvenience, please forgive me. Thank you!

paultaiton commented:
No problem, just wanted to make sure it was not waiting on me. Appreciate the response.

paultaiton commented:
Thanks for your help @haiyuazhang , and especially for your help @Fred-sun . It is much appreciated.
